firstvoices/backend/views/widget_views.py

```diff
@@ -0,0 +1,55 @@
+from django.db.models import Prefetch
+from drf_spectacular.utils import OpenApiResponse, extend_schema, extend_schema_view
+from rest_framework.viewsets import ModelViewSet
+
+from backend.models.widget import SiteWidget, WidgetSettings
+from backend.serializers.widget_serializers import SiteWidgetDetailSerializer
+from backend.views import doc_strings
+from backend.views.api_doc_variables import id_parameter, site_slug_parameter
+from backend.views.base_views import FVPermissionViewSetMixin, SiteContentViewSetMixin
+
+
+@extend_schema_view(
+    list=extend_schema(
+        description="A list of available widgets for the specified site.",
+        responses={
+            200: SiteWidgetDetailSerializer,
+            403: OpenApiResponse(description=doc_strings.error_403_site_access_denied),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+    ),
+    retrieve=extend_schema(
+        description="A widget from the specified site.",
+        responses={
+            200: SiteWidgetDetailSerializer,
+            403: OpenApiResponse(description=doc_strings.error_403_site_access_denied),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
```

Comment:
Super minor but the doc strings for the retrieve errors can be error_403 and error_404 rather than the site-level ones